sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc

```diff
@@ -0,0 +1,976 @@
+some associated scratch space and
+`sycl::ext::oneapi::is_property_list_v<std::decay_t<Properties>>` is true.
+
+_Mandates_: `comp` must satisfy the requirements of `Compare` from the {cpp} standard.
+
+_Effects_: Sort elements in the range containing of elements inside `values` from all work-items
+from the group with respect to the binary comparison function object `comp` using the `gh` group
+helper object.
+Result of sorting is placed into `values` with data placements specified by `properties`.
+Default data placements are those that are specified by the `group_algorithm_data_placement::blocked` property.
+
+_Complexity_: Let `N` be the work-group or sub-group size. `O(N*log(N)*log(N))` comparisons.
+
+4._Constraints_: Only available if `GroupHelper` was created with a work-group or a sub-group and
+some associated scratch space and
+`sycl::ext::oneapi::is_property_list_v<std::decay_t<Properties>>` is true.
+
+_Mandates_: `comp` must satisfy the requirements of `Compare` from the {cpp} standard.
+
+_Effects_: Perform key-value sorting for elements in ranges containing of elements
+inside `keys` and `values` from all work-items from the group with respect to the binary comparison
+function object `comp` using the `gh` group helper object.
+Result of sorting is placed into `keys` and `values` with data placements specified by `properties`.
+Default data placements are those that are specified by the `group_algorithm_data_placement::blocked` property.
+Elements are compared by `operator<`.
+
+_Complexity_: Let `N` be the work-group or sub-group size. `O(N*log(N)*log(N))` comparisons.
+
+5._Constraints_: All functions are available only if `Sorter` is a SYCL Sorter and
+it provides `operator()(Group, sycl::span)` overload and
+`sycl::ext::oneapi::is_property_list_v<std::decay_t<Properties>>` is true.
+
+_Effects_: Equivalent to: `return sorter(g, values, properties)`.
+
+6._Constraints_: All functions are available only if `Sorter` is a SYCL Sorter and
+it provides `operator()(Group, sycl::span, sycl::span)` overload and
+`sycl::ext::oneapi::is_property_list_v<std::decay_t<Properties>>` is true.
+
+_Effects_: Equivalent to: `return sorter(g, keys, values, properties)`.
+
+== Sorters
+
+Sorter is a special type that encapsulates a sorting algorithm. Sorter may contain parameters
+that help to get better performance. Data for sorting are provided to the `operator()`
+that should contain an implementation of a sorting algorithm.
+General semantics of `operator()` is following:
+
+[source,c++]
+----
+template<typename Group, typename Ptr>
+void operator()(Group g, Ptr first, Ptr last);
+
+template<typename Group, typename T>
+T operator()(Group g, T val);
+
+template<typename Group, typename T, std::size_t ItemsPerWorkItem, typename... Properties>
+void operator()(Group g, sycl::span<T, ItemsPerWorkItem> values, sycl::ext::oneapi::experimental::properties properties = {});

```

Comment:
```suggestion
template<typename Group, typename T, std::size_t ItemsPerWorkItem, typename Properties>
void operator()(Group g, sycl::span<T, ItemsPerWorkItem> values);
```

I think you want something like this here.  I don't see why you need to pass a `properties` parameter to this function.  Since it's a compile-time constant property list, isn't it sufficient to have the template parameter?

Reply:
Doing it that way could be OK.  We need to make sure the compiler removes all the overhead, though, when passing the `properties` parameter.  Since the properties are all compile-time, there isn't any data that needs to be passed to the function at runtime.